calculator_v5.rb

```diff
@@ -0,0 +1,93 @@
+  elsif operation == "-" || operation == "subtract" || operation == "minus" || operation == "subtraction" || operation == "subtracting"
+    # subtracts and assigns result to variable
+    result = num1 - num2
+    # outputs formula and result celebration.
+    puts "#{num1} - #{num2} = #{result}"
+    puts "The answer is #{result}! Woohoo!!!"
+
+  # if operation is equal to values indicating multiplication
+  elsif operation == "*" || operation == "x" || operation == "multiply" || operation == "multiplication" || operation == "multiplying"
+    # multiples and assigns result to variable
+    result = num1 * num2
+    # outputs formula and result celebration.
+    puts "#{num1} * #{num2} = #{result}"
+    puts "The answer is #{result}! Woohoo!!!"
+
+  # if operation is equal to values indicating division
+  elsif operation == "/" || operation == "divide" || operation == "division" || operation == "dividing"
+    # outputs error if attempt to divide by 0. Otherwise, compute result.
+    if num2 == 0
+      puts "That's ridiculous! You can't divide by zero! Goodbye now!"
+    else
+      # divides and assigns result to variable
+      result = num1 / num2
+      # outputs formula and result celebration.
+      puts "#{num1} / #{num2} = #{result}"
+      puts "The answer is #{result}! Woohoo!!!"
+    end
+
+  # if operation is equal to values indicating exponent
+  elsif operation == "^" || operation == "**" || operation == "exponents" || operation  == "exponent" || operation == "power"
+    # takes num1 to the power of num2 and assigns result to variable
+    result = num1**(num2)
+    # outputs formula and result celebration.
+    puts "#{num1}^#{num2} = #{result}"
+    puts "The answer is #{result}! Woohoo!!!"
+
+  # if operation is equal to values indicating modulo
+  elsif operation == "%" || operation == "modulus" || operation == "modulo" || operation == "remainder"
+  # calculates remainder of num1 divided by num2 and assigns result to variable
+    result  = num1 % num2
+    # outputs formula and result celebration.
+    puts "#{num1} % #{num2} = #{result}"
+    puts "The answer is #{result}! Woohoo!!!"
+
+  # if other operation value, outputs error
+  else
+    puts "I'm sorry, I don't know how to do #{operation}. Goodbye!"
+  end
+end
+
+# Outputs query for first number, then assigns user entered value to num1
+print "What is the first number? "
+num1 = gets.chomp.delete(",")
+
+# Outputs query for second number, then assigns user entered value to num2
+print "Fantastic! How about the second number? "
+num2 = gets.chomp.delete(",")
+
+numbers = [ "0", "1", "2", "3", "4", "5", "6", "7", "8", "9" ]
+
+# if no operation entered, give error
+if operation.length == 0
+  puts "I don't know what you want me to do with your numbers. Next time enter an operation. Goodbye!"
+elsif num1.length == 0 || !numbers.any? { |number| num1.include?(number) } || num2.length == 0 || !numbers.any? { |number| num2.include?(number) }
```

Comment:
I dig what you're doing with the `include?` block here. It's a pretty inventive way to determine if a user's input is valid. Your condition is saying, essentially "the user's input must be at least one character long and include at least one number."

One blindspot in that validation is strings containing both numerals and letters, like `7even`. I think we could take your validation one step further by saying "the user's input must be at least one character long and __must not__ include any _non-numeric characters_." Similar sentiment (and not that much different, code wise), but controls for an additional edge case.